OloEngine/tests/Rendering/MeshOptimizationTest.cpp

```
@@ -0,0 +1,509 @@
+{
+    auto mesh = MakeQuadMesh();
+    auto originalVertCount = mesh->GetVertices().Num();
+    auto originalIdxCount = mesh->GetIndices().Num();
+
+    MeshOptimization::OptimizeMesh(*mesh);
+
+    EXPECT_EQ(mesh->GetVertices().Num(), originalVertCount);
+    EXPECT_EQ(mesh->GetIndices().Num(), originalIdxCount);
+}
+
+TEST(MeshOptimization, OptimizeMeshPreservesTriangleContent)
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)